### PR TITLE
Refactor SearchTypePopover desktop header

### LINKED ISSUE #1836: Refactor the SearchType popover (desktop) for the new header
## Problem

The new header design only has the SearchTypeButton shown on the desktop widths. 

## Description

It will only show the SearchTypes popup, and not the modal.

## Mockups
Figma: https://www.figma.com/file/KxkZmBNk218V2Fke1GeYCu/New-header-component?node-id=561%3A65396

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1836 by @obulat

## Description

This PR creates a `VHeaderDesktop` component that will be used on widths above `lg`. It refactors the `VSearchTypePopover` to only open the popover (so, it removes the modal). It also refactors the desktop filters button to only open the sidebar.

This PR adds an `IsMinScreenLg` `provide` value in the `default.vue` layout that is set to true only after the layout was mounted. This enables us to have components rendered conditionally (with `v-if`) without getting the server and client mismatch on rendering. 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VHeader/VHeaderFilter.vue [hunk 1]
```diff
@@ -156,7 +156,7 @@ export default defineComponent({
 
     return {
       nodeRef,
-      visibleRef,
+      sidebarVisibleRef: visibleRef,
```
Comment (obulat) [inline src/components/VHeader/VHeaderFilter.vue:159]:
```suggestion
      visibleRef,
```

**Review by zackkrida — CHANGES_REQUESTED** [review]
- The search bar in VHeaderDesktop appears to fill all available space; is that the correct behavior or is it meant to have a max-width?
- I don't think it matters since this will be replaced, but there's a slight alignment problem in a medium breakpoint:
CleanShot 2022-09-22 at 14 20 20@2x
- It occurred to me that I don't think we even need JavaScript to switch between the different headers. A naive CSS approach seems to work fine and avoid the whole hydration mismatch, forced re-render problem.

```vue
<div class="hidden lg:block">
  <VHeaderDesktop />
</div>
<div class="block lg:hidden">
  <VHeader />
</div>
```

**Comment by obulat** [comment]
> * The search bar in VHeaderDesktop appears to fill all available space; is that the correct behavior or is it meant to have a max-width?
> 
>   * I don't think it matters since this will be replaced, but there's a slight alignment problem in a medium breakpoint:

The main goal of this PR is to get rid of the components that have duplicate names, and to separate the two `header` components (mobile and desktop). The styling changes will be fixed in later PRs. Otherwise, this PR will probably be too big.

> - It occurred to me that I don't think we even need JavaScript to switch between the different headers. A naive CSS approach seems to work fine and avoid the whole hydration mismatch, forced re-render problem.
> 
> ```
> <div class="hidden lg:block">
>   <VHeaderDesktop />
> </div>
> <div class="block lg:hidden">
>   <VHeader />
> </div>
> ```

The current solution has the  following downsides:
- hacks on mounted to avoid hydration mismatch.
- forced re-render on desktop widths after `mounted` event is fired. We hope to fix it by having a UI cookie that would make the server render desktop or mobile layout (https://github.com/WordPress/openverse-frontend/milestone/11)

There are a couple of downsides to this CSS solution:
- On the mobile breakpoint, this will be downloading all of the components, both mobile and desktop, and will probably degrade the mobile performance.
- For the search bar components, there will be `id` duplication, or we would have to set separate `id`s for mobile and desktop search inputs.

I was mainly trying to optimize for the mobile experience/performance because mobile usually has lower bandwidths. Do you think it's better to use CSS, @zackkrida, and reduce the need for JS hacks?

**Comment by zackkrida** [comment]
@obulat I think we can follow your solution instead of a CSS approach. We can use the session cookie approach later to fix any visual flashes. Your point about the bundle size is a good one.

**Review by zackkrida — APPROVED** [review]
LGTM and get those components removed.

File: src/components/VLogoLoader/VLogoLoader.vue [hunk 2]
```diff
@@ -5,11 +5,12 @@
     :class="{
       [$style.loading]: status === 'loading' && !prefersReducedMotion,
       'h-10 w-10': autoResize,
+      'h-12 w-12': !autoResize,
     }"
     aria-hidden="true"
     :data-prefers-reduced-motion="prefersReducedMotion"
     data-testid="logo-loader"
-    class="inline-flex h-12 w-12 items-center justify-center rounded p-3 md:h-12 md:w-12"
+    class="inline-flex items-center justify-center rounded p-3 md:h-12 md:w-12"
```
Comment (dhruvkb) [inline src/components/VLogoLoader/VLogoLoader.vue:13]: Do the `md:` height and width classes not need to be moved?

**Review by dhruvkb — APPROVED** [review]
LGTM!

File: src/components/VLogoLoader/VLogoLoader.vue (hunk 2 above)
Comment (obulat) [reply src/components/VLogoLoader/VLogoLoader.vue:13]: Moved to the beginning of the class list? I think the prefixed classes are always at the end.

**Pull request merged**